testing/internal/e2e/helpers.go

```diff
@@ -82,11 +82,26 @@ func RunCommand(command string, opt ...Option) *CommandResult {
 		c2.Stdout = &outbuf
 		c2.Stderr = &errbuf
 
-		c1.Start()
-		c2.Start()
-		c1.Wait()
-		w.Close()
-		c2.Wait()
+		err = c1.Start()
+		if err != nil {
+			return &CommandResult{Err: err}
+		}
+		err = c2.Start()
+		if err != nil {
+			return &CommandResult{Err: err}
+		}
+		err = c1.Wait()
+		if err != nil {
+			return &CommandResult{Err: err}
+		}
+		err = w.Close()
+		if err != nil {
+			return &CommandResult{Err: err}
+		}
+		err = c2.Wait()
+		if err != nil {
+			return &CommandResult{Err: err}
+		}
```

Comment:
I know we discussed briefly yeeting this logic altogether, what made you decide to keep it?

Reply:
Oh... I thought this was what we landed on. Let's chat about this during our meeting.